CONTRIBUTING.md

```diff
@@ -51,5 +51,104 @@ This process can be performed automatically by running `make license-generator`.
 
 Sometimes pull requests take a little while to be merged. This is partially because they often have knock-on effects for the rest of the ecosystem, and partially because we want to give core team members time to review and consider changes thoroughly. Please see the organization's [governance document](https://github.com/purescript/governance) for information about when a pull request may be merged.
 
+## Developer Guide
+
+The following instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+### Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
+
+Update stack's package index before proceeding:
+```
+stack update
+```
+
+### Clone
+
+```
+git clone https://github.com/purescript/purescript.git purescript_compiler
+cd purescript_compiler
+```
+
+### Build
+
+```
+stack build
+```
+
+This will take a while the first time it is run.
+
+### Install
+
+If you'd like to test the build in a local shell, you can run:
+```
+stack exec bash
```

Comment:
When I tried this on my version of stack, v2.5.1, it wouldn't do anything. I had to pass either a script or a heredoc to make it run something.

I also think it weird that this appears under the 'install' header despite being described as 'testing' the build locally.

Reply:
`stack exec bash` should create a sub-shell with the recently-built purs version.

```
➜  complier git:(master) ✗ purs --version
0.14.2

➜  complier git:(master) ✗ stack exec bash

miles@x1c5:~/projects/purescript/complier$ purs --version
0.14.2 [development build; commit: f1953214775945b65ba53ae903b4238c352dcd29 DIRTY]

miles@x1c5:~/projects/purescript/complier$ exit

➜  complier git:(master) ✗ purs --version 
0.14.2
```

Maybe this section could be named something other than 'Install', such as 'Integration Testing'. Perhaps change the original 'Testing` header to 'Unit Testing'.

Reply:
Huh... Seems there's something going on with my terminal. I was able to reproduce most of the above behavior, but the info before the `$` part didn't change after running `stack exec bash`

Reply:
I think the header should be `Exploring and testing a locally compiled version of PureScript` because the purpose of this section is to allow others to just try things and see what happens.

Reply:
Changed heading to `Running a locally-compiled version of PureScript`